usecases/objects/references_delete.go

```diff
@@ -41,8 +41,12 @@ func (m *Manager) DeleteObjectReference(ctx context.Context, principal *models.P
 	defer m.metrics.DeleteReferenceDec()
 
 	deprecatedEndpoint := input.Class == ""
-	if input.Class != "" && strings.Count(string(input.Reference.Beacon), "/") == 3 {
-		toClass, toBeacon, replace, err := m.autodetectToClass(ctx, principal, input.Class, input.Property, input.Reference.Beacon)
+	beacon, err := crossref.Parse(input.Reference.Beacon.String())
+	if err != nil {
+		return &Error{"cannot parse beacon", StatusBadRequest, err}
+	}
```

Comment:
if beacon is used only when `input.Class != ""`, parsing can be moved to `if` block